src/components/GameBoard.js

```diff
@@ -30,22 +31,11 @@ function GameBoard({ muted }) {
 			tempBoard.push(row);
 		}
 
-		//creating a walltile object
-		const wall = new WallTile();
-
 		// drawing the border with walltiles
-		for (let i = 0; i < boardHeight; i++) {
-			tempBoard[i][0] = wall;
-			tempBoard[i][boardWidth - 1] = wall;
-		}
-		for (let j = 0; j < boardWidth; j++) {
-			tempBoard[0][j] = wall;
-			tempBoard[boardHeight - 1][j] = wall;
-		}
-	}
+		tempBoard = createWalls(tempBoard, 0, boardWidth, 0, boardHeight);
+	}, []);
```

Comment:
I'm not very familiar with `useCallback()`, but using the `createWalls()` function without returning a value led to a React warning telling me that I needed to include the `createWalls` function inside the dependency array.

I changed the `createWalls()` function to instead return the updated array, and React no longer needed me to put the function inside the dependency array. Interesting 🤔